### ISSUE TITLE
cisco_asa_show_running_cryprom_map does not parse crypto maps without "set security association lifetime"

### ISSUE THREAD

**Issue by microsoftdealer** [networktocode/ntc-templates#784]
##### ISSUE TYPE
 - Bug Report

##### TEMPLATE USING
cisco_asa_show_running-config_crypto_map.textfsm

##### SAMPLE COMMAND OUTPUT
"""
crypto map WAN1_CMAP 10 match address CMAP_RU16
crypto map WAN1_CMAP 10 set peer 1.1.1.1
crypto map WAN1_CMAP 10 set ikev1 transform-set ESP-AES-256-SHA ESP-AES-192-SHA
crypto map WAN1_CMAP 20 match address CMAP_RU11
crypto map WAN1_CMAP 20 set peer 1.1.1.2
crypto map WAN1_CMAP 20 set ikev1 transform-set ESP-AES-256-SHA ESP-AES-192-SHA
crypto map WAN1_CMAP 30 match address CMAP_RU12
crypto map WAN1_CMAP 30 set peer 1.1.1.3
crypto map WAN1_CMAP 30 set ikev1 transform-set ESP-AES-256-SHA ESP-AES-192-SHA
crypto map WAN1_CMAP 40 match address CMAP_RU17
crypto map WAN1_CMAP 40 set peer 1.1.1.4
crypto map WAN1_CMAP 40 set ikev1 transform-set ESP-AES-256-SHA ESP-AES-192-SHA
crypto map WAN1_CMAP 100 match address CMAP_FR1
crypto map WAN1_CMAP 100 set pfs group5
crypto map WAN1_CMAP 100 set peer 
crypto map WAN1_CMAP 100 set ikev1 transform-set DES-MD5
crypto map WAN1_CMAP 100 set security-association lifetime seconds 3600
crypto map WAN1_CMAP 65535 ipsec-isakmp dynamic SYSTEM_DEFAULT_CRYPTO_MAP
crypto map WAN1_CMAP interface WAN1
crypto map S2S_CMAP 10 match address CMAP_RU17
crypto map S2S_CMAP 10 set peer 1.1.1.5
crypto map S2S_CMAP 10 set ikev1 transform-set ESP-AES-256-SHA ESP-AES-192-SHA
"""

##### SUMMARY
There is an error in template - sometimes there is no "crypto map ${CMAP_NAME} &{SEQ} set security-association lifetime seconds (and yes, there could be not only seconds, but kilobytes)", and as a result, this template just overwrites existing vars with new crypto map lines. Only cryptom map sequences with security-association lines and last will be saved. And yes, there are surely another possible lines, but they are not supported by this template.

##### STEPS TO REPRODUCE
Look at my sample config upwise. There is no lifetime defined in most lines, so, template does not work for me.

##### EXPECTED RESULTS
[LIST OF MULTIPLE SEQUENCES]

##### ACTUAL RESULTS
[{'matched_address': 'CMAP_FR_TEST_VPN',
  'map': 'WAN1_CMAP',
  'seq': '100',
  'pfs': 'group5',
  'peer': '1.1.1.4',
  'ike': 'ikev1',
  'transform': 'DES-MD5',
  'sa': '3600'},
 {'matched_address': 'CMAP_RU16_Access-to-Internet',
  'map': 'S2S_CMAP',
  'seq': '10',
  'pfs': '',
  'peer': '1.1.1.5',
  'ike': 'ikev1',
  'transform': 'ESP-AES-256-SHA',
  'sa': ''}]

##### POSSIBLE FIX
Remove Record from last line, and add "Continue.Record" in first line to record existing vars.
Also, for matching sa lifetime in kilobytes it may be possible to add nev Values instead SA.

Fix without possibility to match kilobytes 
"""
Value MATCHED_ADDRESS (\S+)
Value Required MAP (\S+)
Value Required SEQ (\d+)
Value PFS (group\d|\s*)
Value Required PEER (\S+)
Value IKE (\S+)
Value TRANSFORM (\S+)
Value SA (\d+)

Start
  ^crypto map\s\S+\s\d+\smatch -> Continue.Record
  ^crypto map\s${MAP}\s${SEQ}\smatch\saddress\s${MATCHED_ADDRESS}
  ^crypto map\s\S+\s\d+ set\spfs\s${PFS}
  ^crypto map\s\S+\s\d+ set\speer\s${PEER}
  ^crypto map\s\S+\s\d+ set\s${IKE}\stransform-set\s${TRANSFORM}
  ^crypto map\s\S+\s\d+ set\ssecurity-association\slifetime\sseconds\s${SA}
""

Fix withpossibility to match kilobytes 
"""
Value MATCHED_ADDRESS (\S+)
Value Required MAP (\S+)
Value Required SEQ (\d+)
Value PFS (group\d|\s*)
Value Required PEER (\S+)
Value IKE (\S+)
Value TRANSFORM (\S+)
Value SA_kibs (\d+)
Value SA_secs (\d+)

Start
  ^crypto map\s\S+\s\d+\smatch -> Continue.Record
  ^crypto map\s${MAP}\s${SEQ}\smatch\saddress\s${MATCHED_ADDRESS}
  ^crypto map\s\S+\s\d+ set\spfs\s${PFS}
  ^crypto map\s\S+\s\d+ set\speer\s${PEER}
  ^crypto map\s\S+\s\d+ set\s${IKE}\stransform-set\s${TRANSFORM}
  ^crypto map\s\S+\s\d+ set\ssecurity-association\slifetime\sseconds\s${SA_secs}
  ^crypto map\s\S+\s\d+ set\ssecurity-association\slifetime\skilobytes\s${SA_kibs}
""

**Comment by itdependsnetworks** [comment]
If you have the fix, feel free to put in a PR.

**Comment by diepes** [comment]
@microsoftdealer can you run the asa command showing **all** values (inc. defaults) and show command and output ?

The current output does not seem structured enough to parse with textfsm, and missing defaults will result in a lot of blank's in return table.

**Comment by microsoftdealer** [comment]
@itdependsnetworks , Yes, the fix is up in my first message. I can put in PR, surely, but firstly I need to test it.
@diepes , hadn't understand you, unfortunately. The output in my report is full, provided with "show run crypto map".

**Comment by diepes** [comment]
@microsoftdealer is there an "all" option to the command to display hidden/default values ?

e.g. show running-config crypto map all ?

**Comment by microsoftdealer** [comment]
@diepes , no, ASA doesn't supports this command. I can run "show run all crypto map" which is more verbose, but it will not be executed with textfsm parser of netmiko, cause of the "cisco_asa_show_running-config_crypto_map.textfsm, .*, cisco_asa, sh[[ow]] ru[[nning-config]] cr[[ypto]] m[[ap]]" in index.

**Comment by microsoftdealer** [comment]
@diepes 
`gw-ru13-01/pri/act# show run all crypto map
crypto map WAN1_CMAP 10 match address CMAP_RU16
crypto map WAN1_CMAP 10 set connection-type bidirectional
crypto map WAN1_CMAP 10 set peer 1.1.1.1
crypto map WAN1_CMAP 10 set ikev1 phase1-mode main
crypto map WAN1_CMAP 10 set ikev1 transform-set ESP-AES-256-SHA ESP-AES-192-SHA
crypto map WAN1_CMAP 10 set ikev2 mode tunnel
no crypto map WAN1_CMAP 10 set tfc-packets
crypto map WAN1_CMAP 20 match address CMAP_RU11
crypto map WAN1_CMAP 20 set connection-type bidirectional
crypto map WAN1_CMAP 20 set peer 2.2.2.2
crypto map WAN1_CMAP 20 set ikev1 phase1-mode main
crypto map WAN1_CMAP 20 set ikev1 transform-set ESP-AES-256-SHA ESP-AES-192-SHA
crypto map WAN1_CMAP 20 set ikev2 mode tunnel
no crypto map WAN1_CMAP 20 set tfc-packets
crypto map WAN1_CMAP 30 match address CMAP_RU12
crypto map WAN1_CMAP 30 set connection-type bidirectional
crypto map WAN1_CMAP 30 set peer 3.3.3.3
crypto map WAN1_CMAP 30 set ikev1 phase1-mode main
crypto map WAN1_CMAP 30 set ikev1 transform-set ESP-AES-256-SHA ESP-AES-192-SHA
crypto map WAN1_CMAP 30 set ikev2 mode tunnel
no crypto map WAN1_CMAP 30 set tfc-packets
crypto map WAN1_CMAP 40 match address CMAP_RU17
crypto map WAN1_CMAP 40 set connection-type bidirectional
crypto map WAN1_CMAP 40 set peer 4.4.4.4
crypto map WAN1_CMAP 40 set ikev1 phase1-mode main
crypto map WAN1_CMAP 40 set ikev1 transform-set ESP-AES-256-SHA ESP-AES-192-SHA
crypto map WAN1_CMAP 40 set ikev2 mode tunnel
no crypto map WAN1_CMAP 40 set tfc-packets
crypto map WAN1_CMAP 100 match address CMAP_FR_TEST_VPN
crypto map WAN1_CMAP 100 set pfs group5
crypto map WAN1_CMAP 100 set connection-type bidirectional
crypto map WAN1_CMAP 100 set peer 185.108.69.37
crypto map WAN1_CMAP 100 set ikev1 phase1-mode main
crypto map WAN1_CMAP 100 set ikev1 transform-set DES-MD5
crypto map WAN1_CMAP 100 set ikev2 mode tunnel
crypto map WAN1_CMAP 100 set security-association lifetime seconds 3600
no crypto map WAN1_CMAP 100 set tfc-packets
crypto map WAN1_CMAP 65535 ipsec-isakmp dynamic SYSTEM_DEFAULT_CRYPTO_MAP
crypto map WAN1_CMAP interface WAN1
crypto map S2S_CMAP 10 match address CMAP_RU17
crypto map S2S_CMAP 10 set connection-type bidirectional
crypto map S2S_CMAP 10 set peer 10.0.10.1
crypto map S2S_CMAP 10 set ikev1 phase1-mode main
crypto map S2S_CMAP 10 set ikev1 transform-set ESP-AES-256-SHA ESP-AES-192-SHA
crypto map S2S_CMAP 10 set ikev2 mode tunnel
no crypto map S2S_CMAP 10 set tfc-packets
crypto map S2S_CMAP 20 match address CMAP_RU12
crypto map S2S_CMAP 20 set connection-type bidirectional
crypto map S2S_CMAP 20 set peer 10.0.20.1
crypto map S2S_CMAP 20 set ikev1 phase1-mode main
crypto map S2S_CMAP 20 set ikev1 transform-set ESP-AES-256-SHA ESP-AES-192-SHA
crypto map S2S_CMAP 20 set ikev2 mode tunnel
no crypto map S2S_CMAP 20 set tfc-packets
crypto map S2S_CMAP 30 match address CMAP_RU11
crypto map S2S_CMAP 30 set connection-type bidirectional
crypto map S2S_CMAP 30 set peer 10.0.30.1
crypto map S2S_CMAP 30 set ikev1 phase1-mode main
crypto map S2S_CMAP 30 set ikev1 transform-set ESP-AES-256-SHA ESP-AES-192-SHA
crypto map S2S_CMAP 30 set ikev2 mode tunnel
no crypto map S2S_CMAP 30 set tfc-packets
crypto map S2S_CMAP 40 match address CMAP_RU16
crypto map S2S_CMAP 40 set connection-type bidirectional
crypto map S2S_CMAP 40 set peer 10.0.40.1
crypto map S2S_CMAP 40 set ikev1 phase1-mode main
crypto map S2S_CMAP 40 set ikev1 transform-set ESP-AES-256-SHA ESP-AES-192-SHA
crypto map S2S_CMAP 40 set ikev2 mode tunnel
no crypto map S2S_CMAP 40 set tfc-packets
crypto map S2S_CMAP interface S2SVPN
`

**Comment by microsoftdealer** [comment]
@diepes , here is output with 'all'. Anyway, as I told earlier - it's is not usable with current textfsm command mapping in netmiko.

================
gw-ru13-01/pri/act# show run all crypto map
crypto map WAN1_CMAP 10 match address CMAP_RU16
crypto map WAN1_CMAP 10 set connection-type bidirectional
crypto map WAN1_CMAP 10 set peer 1.1.1.1
crypto map WAN1_CMAP 10 set ikev1 phase1-mode main
crypto map WAN1_CMAP 10 set ikev1 transform-set ESP-AES-256-SHA ESP-AES-192-SHA
crypto map WAN1_CMAP 10 set ikev2 mode tunnel
no crypto map WAN1_CMAP 10 set tfc-packets
crypto map WAN1_CMAP 20 match address CMAP_RU11
crypto map WAN1_CMAP 20 set connection-type bidirectional
crypto map WAN1_CMAP 20 set peer 2.2.2.2
crypto map WAN1_CMAP 20 set ikev1 phase1-mode main
crypto map WAN1_CMAP 20 set ikev1 transform-set ESP-AES-256-SHA ESP-AES-192-SHA
crypto map WAN1_CMAP 20 set ikev2 mode tunnel
no crypto map WAN1_CMAP 20 set tfc-packets
crypto map WAN1_CMAP 30 match address CMAP_RU12
crypto map WAN1_CMAP 30 set connection-type bidirectional
crypto map WAN1_CMAP 30 set peer 3.3.3.3
crypto map WAN1_CMAP 30 set ikev1 phase1-mode main
crypto map WAN1_CMAP 30 set ikev1 transform-set ESP-AES-256-SHA ESP-AES-192-SHA
crypto map WAN1_CMAP 30 set ikev2 mode tunnel
no crypto map WAN1_CMAP 30 set tfc-packets
crypto map WAN1_CMAP 40 match address CMAP_RU17
crypto map WAN1_CMAP 40 set connection-type bidirectional
crypto map WAN1_CMAP 40 set peer 4.4.4.4
crypto map WAN1_CMAP 40 set ikev1 phase1-mode main
crypto map WAN1_CMAP 40 set ikev1 transform-set ESP-AES-256-SHA ESP-AES-192-SHA
crypto map WAN1_CMAP 40 set ikev2 mode tunnel
no crypto map WAN1_CMAP 40 set tfc-packets
crypto map WAN1_CMAP 100 match address CMAP_FR_TEST_VPN
crypto map WAN1_CMAP 100 set pfs group5
crypto map WAN1_CMAP 100 set connection-type bidirectional
crypto map WAN1_CMAP 100 set peer 185.108.69.37
crypto map WAN1_CMAP 100 set ikev1 phase1-mode main
crypto map WAN1_CMAP 100 set ikev1 transform-set DES-MD5
crypto map WAN1_CMAP 100 set ikev2 mode tunnel
crypto map WAN1_CMAP 100 set security-association lifetime seconds 3600
no crypto map WAN1_CMAP 100 set tfc-packets
crypto map WAN1_CMAP 65535 ipsec-isakmp dynamic SYSTEM_DEFAULT_CRYPTO_MAP
crypto map WAN1_CMAP interface WAN1
crypto map S2S_CMAP 10 match address CMAP_RU17
crypto map S2S_CMAP 10 set connection-type bidirectional
crypto map S2S_CMAP 10 set peer 10.0.10.1
crypto map S2S_CMAP 10 set ikev1 phase1-mode main
crypto map S2S_CMAP 10 set ikev1 transform-set ESP-AES-256-SHA ESP-AES-192-SHA
crypto map S2S_CMAP 10 set ikev2 mode tunnel
no crypto map S2S_CMAP 10 set tfc-packets
crypto map S2S_CMAP 20 match address CMAP_RU12
crypto map S2S_CMAP 20 set connection-type bidirectional
crypto map S2S_CMAP 20 set peer 10.0.20.1
crypto map S2S_CMAP 20 set ikev1 phase1-mode main
crypto map S2S_CMAP 20 set ikev1 transform-set ESP-AES-256-SHA ESP-AES-192-SHA
crypto map S2S_CMAP 20 set ikev2 mode tunnel
no crypto map S2S_CMAP 20 set tfc-packets
crypto map S2S_CMAP 30 match address CMAP_RU11
crypto map S2S_CMAP 30 set connection-type bidirectional
crypto map S2S_CMAP 30 set peer 10.0.30.1
crypto map S2S_CMAP 30 set ikev1 phase1-mode main
crypto map S2S_CMAP 30 set ikev1 transform-set ESP-AES-256-SHA ESP-AES-192-SHA
crypto map S2S_CMAP 30 set ikev2 mode tunnel
no crypto map S2S_CMAP 30 set tfc-packets
crypto map S2S_CMAP 40 match address CMAP_RU16
crypto map S2S_CMAP 40 set connection-type bidirectional
crypto map S2S_CMAP 40 set peer 10.0.40.1
crypto map S2S_CMAP 40 set ikev1 phase1-mode main
crypto map S2S_CMAP 40 set ikev1 transform-set ESP-AES-256-SHA ESP-AES-192-SHA
crypto map S2S_CMAP 40 set ikev2 mode tunnel
no crypto map S2S_CMAP 40 set tfc-packets
crypto map S2S_CMAP interface S2SVPN
=============

**Comment by diepes** [comment]
@microsoftdealer  the above is parsable.
"cisco asa show running all cryprom map"  added PR #840 

Output - please check if it is what is expected.

```
---
parsed_sample:
  - matched_address: "CMAP_RU16"
    connection_type: "bidirectional"
    map: "WAN1_CMAP"
    seq: "10"
    pfs: ""
    peer: "1.1.1.1"
    ikev1_phase1_mode: "main"
    ikev1_transform_set: "ESP-AES-256-SHA"
    ikev2_mode: "tunnel"
    isakmp_dynamic: ""
    interface: "WAN1"
    transform: ""
    sa: ""
  - matched_address: "CMAP_RU11"
    connection_type: "bidirectional"
    map: "WAN1_CMAP"
    seq: "20"
    pfs: ""
    peer: "2.2.2.2"
    ikev1_phase1_mode: "main"
    ikev1_transform_set: "ESP-AES-256-SHA"
    ikev2_mode: "tunnel"
    isakmp_dynamic: ""
    interface: "WAN1"
    transform: ""
    sa: ""
  - matched_address: "CMAP_RU12"
    connection_type: "bidirectional"
    map: "WAN1_CMAP"
    seq: "30"
    pfs: ""
    peer: "3.3.3.3"
    ikev1_phase1_mode: "main"
    ikev1_transform_set: "ESP-AES-256-SHA"
    ikev2_mode: "tunnel"
    isakmp_dynamic: ""
    interface: "WAN1"
    transform: ""
    sa: ""
  - matched_address: "CMAP_RU17"
    connection_type: "bidirectional"
    map: "WAN1_CMAP"
    seq: "40"
    pfs: ""
    peer: "4.4.4.4"
    ikev1_phase1_mode: "main"
    ikev1_transform_set: "ESP-AES-256-SHA"
    ikev2_mode: "tunnel"
    isakmp_dynamic: ""
    interface: "WAN1"
    transform: ""
    sa: ""
  - matched_address: "CMAP_FR_TEST_VPN"
    connection_type: "bidirectional"
    map: "WAN1_CMAP"
    seq: "100"
    pfs: "group5"
    peer: "185.108.69.37"
    ikev1_phase1_mode: "main"
    ikev1_transform_set: "DES-MD5"
    ikev2_mode: "tunnel"
    isakmp_dynamic: ""
    interface: "WAN1"
    transform: ""
    sa: "3600"
  - matched_address: "CMAP_RU17"
    connection_type: "bidirectional"
    map: "S2S_CMAP"
    seq: "10"
    pfs: ""
    peer: "10.0.10.1"
    ikev1_phase1_mode: "main"
    ikev1_transform_set: "ESP-AES-256-SHA"
    ikev2_mode: "tunnel"
    isakmp_dynamic: "SYSTEM_DEFAULT_CRYPTO_MAP"
    interface: "WAN1"
    transform: ""
    sa: ""
  - matched_address: "CMAP_RU12"
    connection_type: "bidirectional"
    map: "S2S_CMAP"
    seq: "20"
    pfs: ""
    peer: "10.0.20.1"
    ikev1_phase1_mode: "main"
    ikev1_transform_set: "ESP-AES-256-SHA"
    ikev2_mode: "tunnel"
    isakmp_dynamic: ""
    interface: "S2SVPN"
    transform: ""
    sa: ""
  - matched_address: "CMAP_RU11"
    connection_type: "bidirectional"
    map: "S2S_CMAP"
    seq: "30"
    pfs: ""
    peer: "10.0.30.1"
    ikev1_phase1_mode: "main"
    ikev1_transform_set: "ESP-AES-256-SHA"
    ikev2_mode: "tunnel"
    isakmp_dynamic: ""
    interface: "S2SVPN"
    transform: ""
    sa: ""
  - matched_address: "CMAP_RU16"
    connection_type: "bidirectional"
    map: "S2S_CMAP"
    seq: "40"
    pfs: ""
    peer: "10.0.40.1"
    ikev1_phase1_mode: "main"
    ikev1_transform_set: "ESP-AES-256-SHA"
    ikev2_mode: "tunnel"
    isakmp_dynamic: ""
    interface: "S2SVPN"
    transform: ""
    sa: ""

```

**Comment by diepes** [comment]
@microsoftdealer in your initial message, the "SAMPLE COMMAND OUTPUT"
Has no peer value for this line ?  Was this a copy/paste mistake ?

> crypto map WAN1_CMAP 100 set peer

**Comment by microsoftdealer** [comment]
@diepes, thnx for the PR. What about sa kilobytes lifetime matcing? For now template support only lifetime in seconds.
Yes, it copy/paste mistake, you're right.

**Comment by diepes** [comment]
@microsoftdealer can add match for sa kbytes, please provide command +all, and output from asa containing it +sa_kBytes. ( Needed to add tests to prevent future breaking of template)

sa_seconds & sa_kbytes ?

**Comment by microsoftdealer** [comment]
@diepes, there is no any ASAs with this config in my production. You can replace any of "security-association" lines in previous config with this:

crypto map WAN1_CMAP 100 set security-association lifetime kilobytes 10000000

Also there can be even more complicated thing - seconds OR kilobytes:

crypto map WAN1_CMAP 100 set security-association lifetime seconds 86400 kilobytes 1000000
crypto map WAN1_CMAP 110 set security-association lifetime kilobytes 1000000 seconds 86400

**Comment by diepes** [comment]
@microsoftdealer if we can get a output from a asa device to use as a test scenario, i will update the template.
the tests are there to prevent regressions as templates change.

**Comment by microsoftdealer** [comment]
@diepes, use this sample.

gw-ru13-01/pri/act# show running all crypto map
crypto map WAN1_CMAP 10 match address CMAP_RU16
crypto map WAN1_CMAP 10 set connection-type bidirectional
crypto map WAN1_CMAP 10 set peer 192.0.2.1
crypto map WAN1_CMAP 10 set ikev1 phase1-mode main
crypto map WAN1_CMAP 10 set ikev1 transform-set ESP-AES-256-SHA ESP-AES-192-SHA
crypto map WAN1_CMAP 10 set ikev2 mode tunnel
no crypto map WAN1_CMAP 10 set tfc-packets
crypto map WAN1_CMAP 20 match address CMAP_RU11
crypto map WAN1_CMAP 20 set connection-type bidirectional
crypto map WAN1_CMAP 20 set peer 192.0.2.2
crypto map WAN1_CMAP 20 set ikev1 phase1-mode main
crypto map WAN1_CMAP 20 set ikev1 transform-set ESP-AES-256-SHA ESP-AES-192-SHA
crypto map WAN1_CMAP 20 set ikev2 mode tunnel
crypto map WAN1_CMAP 20 set security-association lifetime seconds 3600
no crypto map WAN1_CMAP 20 set tfc-packets
crypto map WAN1_CMAP 30 match address CMAP_RU12
crypto map WAN1_CMAP 30 set connection-type bidirectional
crypto map WAN1_CMAP 30 set peer 192.0.2.3
crypto map WAN1_CMAP 30 set ikev1 phase1-mode main
crypto map WAN1_CMAP 30 set ikev1 transform-set ESP-AES-256-SHA ESP-AES-192-SHA
crypto map WAN1_CMAP 30 set ikev2 mode tunnel
crypto map WAN1_CMAP 30 set security-association lifetime kilobytes 100000
no crypto map WAN1_CMAP 30 set tfc-packets
crypto map WAN1_CMAP 40 match address CMAP_RU17
crypto map WAN1_CMAP 40 set connection-type bidirectional
crypto map WAN1_CMAP 40 set peer 192.0.2.4
crypto map WAN1_CMAP 40 set ikev1 phase1-mode main
crypto map WAN1_CMAP 40 set ikev1 transform-set ESP-AES-256-SHA ESP-AES-192-SHA
crypto map WAN1_CMAP 40 set ikev2 mode tunnel
crypto map WAN1_CMAP 40 set security-association lifetime kilobytes 100000 seconds 3600
no crypto map WAN1_CMAP 40 set tfc-packets
crypto map WAN1_CMAP 100 match address CMAP_FR_TEST_VPN
crypto map WAN1_CMAP 100 set pfs group5
crypto map WAN1_CMAP 100 set connection-type bidirectional
crypto map WAN1_CMAP 100 set peer 192.0.2.5
crypto map WAN1_CMAP 100 set ikev1 phase1-mode main
crypto map WAN1_CMAP 100 set ikev1 transform-set DES-MD5
crypto map WAN1_CMAP 100 set ikev2 mode tunnel
crypto map WAN1_CMAP 100 set security-association lifetime seconds 3600 kilobytes 100000
no crypto map WAN1_CMAP 100 set tfc-packets
crypto map WAN1_CMAP 65535 ipsec-isakmp dynamic SYSTEM_DEFAULT_CRYPTO_MAP
crypto map WAN1_CMAP interface WAN1
crypto map S2S_CMAP 10 match address CMAP_RU17
crypto map S2S_CMAP 10 set connection-type bidirectional
crypto map S2S_CMAP 10 set peer 172.21.251.10
crypto map S2S_CMAP 10 set ikev1 phase1-mode main
crypto map S2S_CMAP 10 set ikev1 transform-set ESP-AES-256-SHA ESP-AES-192-SHA
crypto map S2S_CMAP 10 set ikev2 mode tunnel
no crypto map S2S_CMAP 10 set tfc-packets
crypto map S2S_CMAP 20 match address CMAP_RU12
crypto map S2S_CMAP 20 set connection-type bidirectional
crypto map S2S_CMAP 20 set peer 172.21.251.26
crypto map S2S_CMAP 20 set ikev1 phase1-mode main
crypto map S2S_CMAP 20 set ikev1 transform-set ESP-AES-256-SHA ESP-AES-192-SHA
crypto map S2S_CMAP 20 set ikev2 mode tunnel
no crypto map S2S_CMAP 20 set tfc-packets
crypto map S2S_CMAP 30 match address CMAP_RU11
crypto map S2S_CMAP 30 set connection-type bidirectional
crypto map S2S_CMAP 30 set peer 172.21.251.18
crypto map S2S_CMAP 30 set ikev1 phase1-mode main
crypto map S2S_CMAP 30 set ikev1 transform-set ESP-AES-256-SHA ESP-AES-192-SHA
crypto map S2S_CMAP 30 set ikev2 mode tunnel
no crypto map S2S_CMAP 30 set tfc-packets
crypto map S2S_CMAP 40 match address CMAP_RU16
crypto map S2S_CMAP 40 set connection-type bidirectional
crypto map S2S_CMAP 40 set peer 172.21.251.34
crypto map S2S_CMAP 40 set ikev1 phase1-mode main
crypto map S2S_CMAP 40 set ikev1 transform-set ESP-AES-256-SHA ESP-AES-192-SHA
crypto map S2S_CMAP 40 set ikev2 mode tunnel
no crypto map S2S_CMAP 40 set tfc-packets
crypto map S2S_CMAP 160 match address CMAP_RU16_Access-to-Internet
crypto map S2S_CMAP 160 set connection-type bidirectional
crypto map S2S_CMAP 160 set peer 172.21.251.34
crypto map S2S_CMAP 160 set ikev1 phase1-mode main
crypto map S2S_CMAP 160 set ikev1 transform-set ESP-AES-256-SHA ESP-AES-192-SHA
crypto map S2S_CMAP 160 set ikev2 mode tunnel

**Comment by diepes** [comment]
@microsoftdealer 2nd PR to parse new example output.
https://github.com/networktocode/ntc-templates/pull/883
Split SA into SA_SEC and SA_KB

Please have a look at the PR https://github.com/networktocode/ntc-templates/pull/883 and the generated parsing for the test.

**Comment by microsoftdealer** [comment]
@diepes , looks pretty good!

**Comment by FragmentedPacket** [comment]
Closed via #883
